src/platform.node.ts

```diff
@@ -52,15 +62,54 @@ export class InputFile {
      * @param filename Optional name of the file
      */
     constructor(
-        file: string | Uint8Array | ReadStream | AsyncIterable<Uint8Array>,
+        file:
+            | string
+            | URL
+            | URLLike
+            | Uint8Array
+            | ReadStream
+            | AsyncIterable<Uint8Array>,
         filename?: string,
     ) {
-        this[inputFileData] = file;
+        this.fileData = file;
         if (filename === undefined && typeof file === "string") {
             filename = basename(file);
         }
         this.filename = filename;
     }
+    get [inputFileData]() {
+        if (this.consumed) {
+            throw new Error("Cannot reuse InputFile data source!");
+        }
+        let data = this.fileData;
+        if (
+            typeof data === "object" && ("url" in data || data instanceof URL)
+        ) {
+            data = fetchFile(data instanceof URL ? data : data.url);
+        } else if (
+            typeof data !== "string" && (!(data instanceof Uint8Array))
+        ) {
+            this.consumed = false;
+        }
+        return data;
+    }
+}
+
+async function* fetchFile(url: string | URL): AsyncIterable<Uint8Array> {
+    const { body } = await fetch(url);
```

Comment:
is `fetch` global? i don't see where it come from

Reply:
Only on Deno and the web. On Node, it is shimmed by deno2node using node-fetch.

Reply:
Maybe it should be improved in deno2node that the shimming isn't performed for Node-specific source files.

Reply:
I don't know anything about deno2node. Just looking at the file as a whole from Node.js dev perspective i see `fetch` that is not imported from anywhere so it will crash until `fetch` is added to globals

Reply:
I can only do that after the changes to deno2node. Right now, the shimming will produce clashing imports, hence causing the compilation to fail.

Reply:
Ah I see. No, shimming in deno2node works by injecting import statements into the modules that actually need the shim. It does not touch global definitions.

I wrote a new README in https://github.com/wojpawlik/deno2node/pull/14 a long time ago but it has not been merged yet. I improved this README now with respect to this misunderstanding. I hope we have a chance to get this through soon

cc @wojpawlik